packages/core/src/StateMachine.ts

```diff
@@ -328,7 +328,12 @@ export class StateMachine<
       undefined
     );
     nextState.actions.unshift(...preInitialState.actions);
-    return macrostep(nextState, null as any, this) as StateFrom<typeof this>;
+
+    const macroState = macrostep(nextState, null as any, this) as StateFrom<
+      typeof this
+    >;
+    macroState.changed = undefined;
```

Comment:
`.changed === undefined` somewhat means the initial state but it wasn't delivering on that promise here - because `macrostep` was already computing a boolean for it as the `nextState` here already didn't have `._initial` set on it - that's only available on the `preInitialState` that was already "changed" by the `resolveMicroTransition` just 2 lines above this

sideways... i'd really like to explore removing `.changed` altogether